### PR TITLE
Improve trailing comma situation

### PR DESCRIPTION
**Pull request by windelbouwman**

### CODE REVIEW

File: parser/src/python.lalrpop [hunk 1]
```diff
@@ -817,7 +795,7 @@ Atom: ast::Expression = {
         // List comprehension:
         e
     },
-    "(" <e:TestList?> <trailing_comma:","?> ")" => {
+    "(" <e:TestList2?> <trailing_comma:","?> ")" => {
```
Comment (adrian17) [inline parser/src/python.lalrpop]: Why do we need this part in the first place?
`TestList` already parses `1, 2, 3` as a tuple, and wrapping it in `()` should be a no-op, like any other expression.

This seems to work for me (passes all tests, didn't look for edge cases):
```
    "(" ")" => ast::Expression::Tuple { elements: vec![] },
    "(" <e:TestList> ")" => { e },
```
And then (and after resolving the issue above) we can drop `TestList2`.

File: parser/src/python.lalrpop [hunk 2]
```diff
@@ -102,7 +102,7 @@ ExpressionStatement: ast::LocatedStatement = {
           }
         }
     },
-    <loc:@L> <expr:TestOrStarExprList> <op:AugAssign> <e2:TestList> => {
+    <loc:@L> <expr:TestOrStarExprList> <op:AugAssign> <e2:TestList2> => {
```
[the remark this answers is not in the document]
Comment (windelbouwman) [reply parser/src/python.lalrpop]: I added a testcase for this. Simply using TestList works.

File: parser/src/python.lalrpop (hunk 1 above)
Comment (windelbouwman) [reply parser/src/python.lalrpop]: I resolved this, and dropped TestList2. Thank you for the review!

File: parser/src/ast.rs [hunk 3]
```diff
@@ -47,7 +47,7 @@ pub enum Statement {
     Break,
     Continue,
     Return {
-        value: Option<Vec<Expression>>,
+        value: Option<Box<Expression>>,
```
Comment (adrian17) [inline parser/src/ast.rs]: couldn't this be simply an `Option<Expression>`?

Comment (windelbouwman) [reply parser/src/ast.rs]: Maybe, it's mainly done due to memory size of the different enum variants that `Box` is present. Maybe `Option` also acts as a sort of Box like pointer to reduce the full size of `Return`.

Comment (adrian17) [reply parser/src/ast.rs]: Well, other Statement variants already hold a non-boxed Expression, so that shouldn't matter here. IMO either all or none (unless later performance measurements prove otherwise).
